### PR TITLE
Add endpoint to be able to fetch more detailed product information for a member

### PR DESCRIPTION
**Pull request by vonElfvin**

### CODE REVIEW

File: src/main/java/com/hedvig/productPricing/service/service/InsuranceBillingServiceImpl.java [hunk 1]
```diff
@@ -266,6 +259,26 @@ public InsuranceBillingDTO getMonthlySubscriptionByMemberId(YearMonth period, St
...
+    public List<InsuranceProductSubscriptionDTO> getMonthlyProductSubscriptionsByMemberId(String memberId, YearMonth period) {
+      val date = LocalDateTime.now().with(period).withDayOfMonth(1).plusMonths(1);
+      List<ProductEntity> productList = productRepository.findBillingByDateByMemberId(date, memberId);
+      return calculateProductSubscription(productList, period);
+    }
+
+    private List<InsuranceProductSubscriptionDTO> calculateProductSubscription(List<ProductEntity> productList, YearMonth period) {
+      Stream<InsuranceBillingInfo> filteredList = productList
+        .stream()
+        .filter(x -> isSignedInsurance(x))
```
Comment (palmfjord) [inline src/main/java/com/hedvig/productPricing/service/service/InsuranceBillingServiceImpl.java]: no `x`s please :D makes it super-hard to understand what we're checking here

File: src/main/java/com/hedvig/productPricing/service/service/InsuranceBillingServiceImpl.java [hunk 2]
```diff
@@ -266,6 +259,26 @@ public InsuranceBillingDTO getMonthlySubscriptionByMemberId(YearMonth period, St
...
+        .map(InsuranceBillingInfo::new);
+
+      return filteredList.map(info -> new InsuranceProductSubscriptionDTO(
```
Comment (palmfjord) [inline src/main/java/com/hedvig/productPricing/service/service/InsuranceBillingServiceImpl.java]: just some nitty-gritty but can we move `.map` and `.collect` to newlines? Makes it sliiightly easier to read

File: src/main/java/com/hedvig/productPricing/service/service/InsuranceBillingServiceImpl.java [hunk 3]
```diff
@@ -266,6 +259,26 @@ public InsuranceBillingDTO getMonthlySubscriptionByMemberId(YearMonth period, St
 
     }
 
+    @Override
+    public List<InsuranceProductSubscriptionDTO> getMonthlyProductSubscriptionsByMemberId(String memberId, YearMonth period) {
+      val date = LocalDateTime.now().with(period).withDayOfMonth(1).plusMonths(1);
```
Comment (palmfjord) [inline src/main/java/com/hedvig/productPricing/service/service/InsuranceBillingServiceImpl.java]: I don't understand this.
a) As i understand it this'll be _next_ month no? Is this expected? It doesn't make sense imo to calculate the price not for the period we're passing but for the period _after_ it, i would expect `getMonthlyProductSubscriptionsByMemberId` to calculate on whatever it receives
b) Why are we using `LocalDateTime.now()`? Why not just `period.atDay(1)` if we want a locadate?

File: src/main/java/com/hedvig/productPricing/service/web/dto/InsuranceProductSubscriptionDTO.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+@Value
+public class InsuranceProductSubscriptionDTO {
+  UUID productId;
+  ProductTypes productType;
+  MonetaryAmount subscription;
+
+  public InsuranceProductSubscriptionDTO(UUID productId, ProductTypes productType, BigDecimal amount) {
+    this.productId = productId;
+    this.productType = productType;
+    this.subscription = Money.of(amount, "SEK");
```
Comment (palmfjord) [inline src/main/java/com/hedvig/productPricing/service/web/dto/InsuranceProductSubscriptionDTO.java]: Why do we need this? I feel like we could just pass `MonetaryAmount` directly and then we don't need to write this constructor at all

File: src/main/java/com/hedvig/productPricing/service/service/InsuranceBillingServiceImpl.java (hunk 3 above)
Comment (vonElfvin) [reply src/main/java/com/hedvig/productPricing/service/service/InsuranceBillingServiceImpl.java]: Honestly I don't know, I'm just doing as the old methods do it 🙈

Comment (palmenhq) [reply src/main/java/com/hedvig/productPricing/service/service/InsuranceBillingServiceImpl.java]: That's a terrible excuse 😬 🙈  this is quite critical I believe, as we must charge for the correct month no?

Comment (vonElfvin) [reply src/main/java/com/hedvig/productPricing/service/service/InsuranceBillingServiceImpl.java]: Given that, as you stated, this is critical, Why would I want to change the date? From my understanding this is thoroughly tested and should this not be tampered with. So for a) no it's most likely the correct date and thus month given this is what we have used all along. And for b) sure that would make it prettier.

Comment (vonElfvin) [reply src/main/java/com/hedvig/productPricing/service/service/InsuranceBillingServiceImpl.java]: I can double check with Meletis though before resolving this.

**Pull request merged**